lib/eslint-patch.js

```diff
@@ -1,33 +1,31 @@
-'use strict';
-var meld = require('meld');
-var docs = require('./docs');
-var Config = require("eslint/lib/config");
-var ConfigUpgrader = require('./config_upgrader');
-
-var supportedPlugins = ['react', 'babel'];
-
-module.exports = function patcher(eslint) {
-
-  meld.around(eslint.CLIEngine, 'loadPlugins', function(joinPoint) {
-    var pluginNames = joinPoint.args[0];
-    var filteredPluginNames = pluginNames.filter(function(pluginName) {
-      return supportedPlugins.indexOf(pluginName) >= 0;
-    });
-    return joinPoint.proceed(filteredPluginNames);
-  });
-
-  meld.around(eslint.CLIEngine, 'addPlugin', function() {
-    return;
-  });
-
-  // meld.around(eslint.CLIEngine.Config, 'loadPackage', function(joinPoint) {
-  //   var filePath = joinPoint.args[0];
-  //   if (filePath.match(/^eslint-config-airbnb.*/)) {
-  //     return joinPoint.proceed();
-  //   } else {
-  //     return {};
-  //   }
-  // });
+"use strict";
+
+const Plugins = require("eslint/lib/config/plugins");
+
+const Config = require("eslint/lib/config");
+const ConfigUpgrader = require("./config_upgrader");
+
+const docs = require("./docs");
+
+module.exports = function patch(eslint) {
+
+  Plugins.loadAll = function(pluginNames) {
+    const loadedPlugins = Object.keys(Plugins.getAll());
+    if(loadedPlugins.length > 0) {
+      return;
+    }
+
+    for(const index in pluginNames) {
+      const name = pluginNames[index];
+      try {
+
+        Plugins.load(name);
+
+      } catch(e) {
+        console.error(`[DEBUG] Plugin ${name} not supported`);
```

Comment:
@GordonDiggs I've just noticed that in the issue #192 you mentioned we should tell the user we're *skipping* the plugin, should we add something on those lines?